include/evmc/evmc.h

```diff
@@ -503,36 +503,111 @@ typedef evmc_bytes32 (*evmc_get_storage_fn)(struct evmc_host_context* context,
  * notation is used:
  * - 0 is zero value,
  * - X != 0 (X is any value other than 0),
- * - Y != X, Y != 0 (Y is any value other than X and 0),
- * - Z != Y (Z is any value other than Y),
- * - the "->" means the change from one value to another.
+ * - Y != 0, Y != X,  (Y is any value other than X and 0),
+ * - Z != 0, Z != X, Z != X (Z is any value other than Y and X and 0),
+ * - the "o -> c -> v" triple describes the change status in the context of:
+ *   - o: original value (cold value before a transaction started),
+ *   - c: current storage value,
+ *   - v: new storage value to be set.
+ *
+ * The order of elements follows EIPs introducing net storage gas costs:
+ * - EIP-2200: https://eips.ethereum.org/EIPS/eip-2200,
+ * - EIP-1283: https://eips.ethereum.org/EIPS/eip-1283.
  */
 enum evmc_storage_status
 {
     /**
-     * The value of a storage item has been left unchanged: 0 -> 0 and X -> X.
+     * The value of a storage item has been left unchanged has been modified again
+     * in the transaction context. This is the group of "no-op" cases related to
+     * minimal gas cost.
+     * 0|X   -> 0 -> 0 (current value unchanged)
+     * 0|X|Y -> Y -> Y (current value unchanged)
+     * 0|X   -> Y -> Z (modified previously added/modified value)
+     *
+     * This corresponds to the EIP-2200 rule 2 where:
+     * c == v.
+     */
+    EVMC_STORAGE_MODIFIED_AGAIN = 0,
```

Comment:
Why `EVMC_STORAGE_MODIFIED_AGAIN` rather than `EVMC_STORAGE_UNCHANGED`?

Reply:
It felt more correct to me: in general we modify a value, just gas cost does not care. The "unchanged" strongly suggests that `current == value` what is not always the case for this group. But I can change it to `EVMC_STORAGE_UNCHANGED` if people think this is fine.

Reply:
What is this rule 2 of EIP-2200 referenced in the doc? "If _current value_ equals _new value_ (this is a no-op), `SLOAD_GAS` is deducted."? Or something else?

Reply:
Oh, this is a mistake. This a group of this rule (2) and 3.2 without matching any subrules of 3.2. If you can parse my DSL, this can be e.g. `X → Y → Z`. Because these have the same cost, I decided to use single enum value for it.

If you think this is too confusing, I can assign separate values to these. But this suggests these have different behavior what is not the case.

Reply:
I'd prefer separate `EVMC_STORAGE_UNCHANGED` and `EVMC_STORAGE_MODIFIED_AGAIN` for extra clarity, even if they are priced the same.

Reply:
I reviewed my work on this and distinguishing these two requires additional check `current == new` (not needed to distinguish all other cases). Moreover, this would be inconvenient to test: e.g. state tests are not able to spot a mistake.

So I'm against.

Reply:
OK, then please amend the comment to say that it covers both rule 2 as well as 3.2 without any of the sub-rules.

Reply:
Please see the changed version.

Reply:
What about calling this guy `EVMC_STORAGE_NO_OP`?

Reply:
I think `NO_OP` may be ok, however the spec uses this term in `current == value` context only.

Another possible name can be `UNAFFECTED`.

Reply:
`UNAFFECTED` sounds good.

Reply:
`REASSIGNED`?

Time for twitter pool.

Reply:
`REASSIGNED` is fine with me.

Reply:
Based on the comments from @gumb0 and my preferences, can we leave it as `EVMC_STATUS_MODIFIED_AGAIN`? Also documentation is updated.

Reply:
Changed to `EVMC_STORAGE_ASSIGNED`.